packages/ui/src/components/ToggleSection/ToggleSection.tsx

```diff
@@ -0,0 +1,294 @@
+import React from "react";
+import { Slot } from "@radix-ui/react-slot";
+import { useControllableState } from "@radix-ui/react-use-controllable-state";
+import * as TogglePrimitive from "@radix-ui/react-toggle";
+
+import Heading, { HeadingProps } from "../Heading";
+
+import {
+  ToggleSectionProvider,
+  useToggleSectionContext,
+} from "./ToggleSectionProvider";
+
+// Names used for unique IDs
+const NAME = {
+  INITIAL_CONTENT: "InitialContent",
```

Comment:
It seems like the design call for three state: Initial, Display, Editable

Reply:
I did see that and had a hard time understanding the real difference between initial and display. Let me know if my assumption was correct:

- **Initial**: Effectively a null state
- **Display**: Primary content that is displaced before "opening" the component (could be null or display some info)
- **Editable**: Secondary content that appears when the component is "opened"

So, my understanding was that initial and display are just two different states that could appear in the primary content and not something the user can control through a toggle.

Reply:
Yeah, open/close is what we need in this component then.